File: 프로그래머스/LV1/명예의 전당.py
def solution(k, score):
    answer = []
    goat = []

    for i in range(len(score)):
        goat.append(score[i])
        goat.sort(reverse=True)

        if len(goat) > k:
            goat.pop()

        answer.append(goat[-1])

    return answer

File: 프로그래머스/javascript/LV1/명예의 전당.js
function solution(k, score) {
  let answer = [];
  let crown = [];
  for (let i = 0; i < score.length; i++) {
    if (crown.length < k) {
      crown.push(score[i]);
    } else {
      if (crown[0] < score[i]) {
        crown[0] = score[i];
      }
    }
    crown.sort((a, b) => a - b);
    answer.push(crown[0]);
  }
  return answer;
}
